src/main/daml/Daml/Finance/Settlement/RouteProvider/Intermediaries.daml

```diff
@@ -0,0 +1,41 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Settlement.RouteProvider.Intermediaries where
+
+import DA.List (groupOn, head)
+import DA.Map qualified as M (Map, lookup)
+import DA.Optional (fromSomeNote)
+import Daml.Finance.Interface.Settlement.RouteProvider qualified as RouteProvider (Discover(..), I, View(..))
+import Daml.Finance.Interface.Types.Common (Parties)
+import Daml.Finance.Settlement.Hierarchy (Hierarchy(..), unfoldStep)
+
+-- | Template which implements the `RouteProvider` interface.
+-- It is used to discover the settlement route for each settlement `Step`, i.e., `RoutedSteps`\s.
+-- For each instrument to settle as part of the batch, a hierarchy of intermediaries is specified
+-- in `paths`. This hierarchy is used to generate the `RoutedStep`\s.
+template RouteProviderWithIntermediaries
```

Comment:
And this `Intermediated`? or `IntermediatedStatic` (as opposed to a possible `IntermediatedDynamic` that would fetch routes / route pieces by key from contracts on the ledgeR)?

Reply:
Renamed `RouteProviderWithIntermediaries` to `IntermediatedStatic`.